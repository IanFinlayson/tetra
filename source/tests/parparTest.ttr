def main():
   z = 0
   print("In main(), just set z = 0 and z = ", z, "\n")
   y = 0
   print("In main(), just set y = 0 and y = ", y, "\n")
   w = 0
   print("In main(), just set w = 0 and w = ", w, "\n")
   n = 0
   print("In main(), just set n = 0 and n = ", n, "\n")
   parallel:
      parallel:
         w = 0
      n = 0
   print("Just passed the the first set of nested parallels and w = ", w, " and n = ", n, "\n")
   parallel:
      if true:
         x = 0
         while x < 1000:
            x+=1
         lock xl:
            pass
      parallel:
         z = 0
         y = 0
         lock xl:
            pass
   print("At the end of main() and x = ", x, ", y = ", y, ", and z = ", z, "\n")
