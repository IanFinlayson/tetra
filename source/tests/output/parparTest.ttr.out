In main(), just set z = 0 and z = 0
In main(), just set y = 0 and y = 0
In main(), just set w = 0 and w = 0
In main(), just set n = 0 and n = 0
Just passed the the first set of nested parallels and w = 0 and n = 0
At the end of main() and x = 1000, y = 0, and z = 0

